autoPyTorch/pipeline/components/training/trainer/base_trainer.py

```diff
@@ -318,8 +320,14 @@ def on_epoch_end(self, X: Dict[str, Any], epoch: int) -> bool:
                 if self.use_snapshot_ensemble:
                     assert self.model_snapshots is not None, "model snapshots container can't be " \
                                                              "none when snapshot ensembling is enabled"
-                    model_copy = deepcopy(self.swa_model) if self.use_stochastic_weight_averaging \
-                        else deepcopy(self.model)
+                    if epoch == self.budget_tracker.max_epochs:
+                        if self.use_stochastic_weight_averaging:
+                            model_copy = deepcopy(self.swa_model)
+                        else:
+                            model_copy = deepcopy(self.model)
+                    else:
+                        model_copy = deepcopy(self.model)
```

Comment:
Could you explain how this could be beneficial?

Reply:
The benefit would be the code cleanup. It does look nicer to me, the only thing I would suggest would be changing the order of`self.use_stochastic_weight_averaging` with `is_last_epoch`

Reply:
Yes, as Arlind mentioned, it reduces the nest of the code block.
Based on his suggestion, this will be the new suggestion.
```suggestion
                    is_last_epoch = (epoch == self.budget_tracker.max_epochs)
                    if is_last_epoch and self.use_stochastic_weight_averaging:
                        model_copy = deepcopy(self.swa_model)
                    else:
                        model_copy = deepcopy(self.model)
```